lib/rules/jsx-sort-default-props.js

```diff
@@ -172,6 +173,14 @@ module.exports = {
           return;
         }
 
+        if (
+          node.parent.type !== 'BinaryExpression'
+          && node.parent.type !== 'AssignmentExpression'
+          && node.parent.type !== 'LogicalExpression'
+        ) {
```

Comment:
tests?

Reply:
My apologies for the confusion. I mistakenly performed unnecessary type narrowing, so I removed it in commit d59ac20. Since the existing test cases seem to be unaffected, I did not add any additional tests.